### ISSUE TITLE
Tips for training the stress model

### ISSUE THREAD

**Issue by zhuomanliu** [HimangiM/UniPhy_CVPR2025#4]
Hi,

Thanks for your awesome work! I've successfully run the Gradio demo. I then tried to run the script train_latent_space.py for training, but I found that the predicted stress values were unstable and appeared to have unreasonable values. Would you have any tips for training a good model?  Many thanks!

(This run followed the network settings (hidden_dim=4096, embed_dim=1024) of the pretrained checkpoint. I also tried using the default settings (hidden_dim=128, embed_dim=128) for training, but the stress prediction remained unstable.)
> Dataset size:  150000000 Number of trajectories:  150
Epoch:  0
[2025-09-09 05:25:26,446][root][INFO] - Epoch: 0, Iter/Total: 0/35645, Loss: 0.004232 (l2: 0.000031, l2_fproj: 0.001000, reg: 0.003201), LR1: [0.0001, 0.001]
[2025-09-09 05:25:35,097][root][INFO] - Epoch: 0, Iter/Total: 100/35645, Loss: 25789.669922 (l2: 25789.666016, l2_fproj: 0.000001, reg: 0.003025), LR1: [0.0001, 0.001]
[2025-09-09 05:25:43,755][root][INFO] - Epoch: 0, Iter/Total: 200/35645, Loss: 9460.906250 (l2: 9460.903320, l2_fproj: 0.000000, reg: 0.002853), LR1: [0.0001, 0.001]
[2025-09-09 05:25:52,267][root][INFO] - Epoch: 0, Iter/Total: 300/35645, Loss: 0.007632 (l2: 0.004899, l2_fproj: 0.000049, reg: 0.002683), LR1: [0.0001, 0.001]
[2025-09-09 05:26:00,849][root][INFO] - Epoch: 0, Iter/Total: 400/35645, Loss: 237147.828125 (l2: 237147.828125, l2_fproj: 0.000000, reg: 0.002518), LR1: [0.0001, 0.001]
[2025-09-09 05:26:09,463][root][INFO] - Epoch: 0, Iter/Total: 500/35645, Loss: 8.700119 (l2: 8.695848, l2_fproj: 0.001915, reg: 0.002355), LR1: [0.0001, 0.001]
[2025-09-09 05:26:17,998][root][INFO] - Epoch: 0, Iter/Total: 600/35645, Loss: 816363.937500 (l2: 816363.937500, l2_fproj: 0.000000, reg: 0.002197), LR1: [0.0001, 0.001]
[2025-09-09 05:26:26,529][root][INFO] - Epoch: 0, Iter/Total: 700/35645, Loss: 291149.406250 (l2: 291149.406250, l2_fproj: 0.000000, reg: 0.002042), LR1: [0.0001, 0.001]
[2025-09-09 05:26:35,095][root][INFO] - Epoch: 0, Iter/Total: 800/35645, Loss: 0.049655 (l2: 0.047763, l2_fproj: 0.000001, reg: 0.001891), LR1: [0.0001, 0.001]
[2025-09-09 05:26:43,739][root][INFO] - Epoch: 0, Iter/Total: 900/35645, Loss: 20999618.000000 (l2: 20999618.000000, l2_fproj: 0.000000, reg: 0.001745), LR1: [0.0001, 0.001]
[2025-09-09 05:26:52,348][root][INFO] - Epoch: 0, Iter/Total: 1000/35645, Loss: 12.203482 (l2: 12.201849, l2_fproj: 0.000030, reg: 0.001603), LR1: [0.0001, 0.001]
......

**Comment by HimangiM** [comment]
Hi, 

Thank you for your interest in our work. When training neural networks to predict stress values, there is a behavior of having high stress values and it takes some epochs to stabilize. It usually happens when the training batch contain targets with very different stress values scale, so the model seems stable on some batches but then is affected by the outliers. It is common with stress especially as stress magnitude can span in orders of magnitude. For a stable training, you can try using a lower learning rate or gradient clipping.
